### PR TITLE
Third assignment

### PR DESCRIPTION
**Pull request by Anastasiiasyvak**

### CODE REVIEW

File: pa3.sql [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    FROM employee_service
+    WHERE id_service IN (
+        SELECT id
+        FROM service
+        WHERE name = 'Massage'
+    )
+);
+
+-- вибираємо всі записи з таблиці appointment, де id_employee відповідає працівникам, які мають позицію "Master of hair styling"
+SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
SELECT <set of columns>
```

File: pa3.sql [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+);
+
+
+-- 4) EXISTS with non-correlated subqueries result
+
+-- Вибираємо всіх клієнтів, які мають хоча б один запис про призначений їм запис
+SELECT *
+FROM customer
+WHERE EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    FROM appointment
+    WHERE appointment.id_customer = customer.id
+);
+
+-- Видаляємо всі записи про послуги, які не мають відповідних записів у таблиці services2products
+DELETE FROM service
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+    FROM services2products
+    WHERE services2products.service_id = service.id
+);
+
+-- Збільшуємо зарплату всім працівникам, які надають послугу "Massage" на 10%
+UPDATE employee
+SET salary = salary * 1.10
+WHERE EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+    INNER JOIN service  ON employee_service.id_service = service.id
+    WHERE employee_service.id_employee = employee.id
+    AND service.name = 'Massage'
+);
+
+-- 5) NOT EXISTS with non-correlated subqueries result
+
+
+-- Вибираємо всіх клієнтів, які не зробили жодного платежу
+SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
SELECT <set of columns>
```

File: pa3.sql [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+FROM customer
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+    FROM appointment
+    WHERE appointment.id_customer = customer.id
+);
+
+-- Оновлюємо статуси всіх записів у таблиці appointment, які не мають відповідних записів у таблиці payment на 'cancelled'
+UPDATE appointment
+SET status = 'cancelled'
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+    FROM payment
+    WHERE payment.id_appointment = appointment.id
+);
+
+-- Видаляємо всі записи про клієнтів, які не зробили жодного платежу
+DELETE FROM customer
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+);
+
+
+-- 9) EXISTS with correlated subqueries result
+
+-- Вибірка співробітників, які надають послугу "Manicure" і мають зустрічі
+SELECT id, name, phone, email, position
+FROM employee
+WHERE EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+
+
+-- 10) NOT EXISTS with correlated subqueries result
+
+
+-- Вибираємр співробітників, які не мають зустрічей
+SELECT id, name, position
+FROM employee
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+    FROM customer
+    WHERE customer.id = appointment.id_customer AND customer.status = 'no show'
+) AND status = 'pending';
+
+
+-- Видалення співробітників, які не мають зустрічей
+DELETE FROM employee
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 12]
```diff
@@ -0,0 +1,347 @@
...
+    FROM appointment
+    WHERE appointment.id_employee = employee.id
+);
+
+
+-- Оновлюємо зустрічі зі статусом "pending", які не мають клієнтів
+UPDATE appointment
+SET status = 'cancelled'
+WHERE status = 'pending' AND NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

File: pa3.sql [hunk 13]
```diff
@@ -0,0 +1,347 @@
...
+    FROM customer
+    WHERE customer.id = appointment.id_customer
+);
+
+
+-- Видаляємо співробітників, які не мають зустрічей
+DELETE FROM employee
+WHERE NOT EXISTS (
+    SELECT *
```
Comment (Ivashchyk) [inline pa3.sql]:
```suggestion
    SELECT 1
```

**Review by Ivashchyk — CHANGES_REQUESTED** [review]
min issues

**Comment by Anastasiiasyvak** [comment]
https://github.com/Anastasiiasyvak/DatabaseBeautySalon/pull/2

**Pull request merged**